src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CookieHelper.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Net.Http
+{
+    internal static class CookieHelper
+    {
+        public static void ProcessReceivedCookies(HttpResponseMessage response, CookieContainer cookieContainer)
+        {
+            IEnumerable<string> setCookies;
+            if (response.Headers.TryGetValues(HttpKnownHeaderNames.SetCookie, out setCookies))
```

Comment:
Nit: based on internal knowledge, do we know anything about what the actual type of setCookies will be?  I'm wondering if, for example, it's always an array, we could cast to the array and avoid the enumerator allocation.

Reply:
Yes, it's always a string[].  I'm a little queasy about doing the cast; seems like it would be better to have an internal API that exposes it as string[].  That said, we are doing this in other places already.